File: 16.c
#include <stdio.h>

unsigned long long int power(int base, int exponent);
int add_digits(unsigned long long int number);

int main()
{
    // printf("Sum: %d\n", add_digits(power(2, 1000)));
    printf("Sum: %llu\n", power(2, 50));
    return 0;
}

unsigned long long int power(int base, int exponent)
{
    unsigned long long int powered = 1;
    for (int i = 0; i < exponent; i++)
    {
        powered *= base;
    }
    return powered;
}
int add_digits(unsigned long long int number)
{
    int sum = 0;
    while (number != 0)
    {
        sum += number % 10;
        number /= 10;
    }
    return sum;
}

File: _16.py
# print(sum([int(d) for d in [d for d in str(pow(2, 1000))]]))


def main():
    print(f"Sum: {add_digits(power(2, 1000))}")


def power(base, exponent):
    powered = 1
    for _ in range(exponent):
        powered *= base
    return powered


def add_digits(number):
    sum = 0
    while number != 0:
        sum += number % 10
        number //= 10

    return sum


if __name__ == "__main__":
    main()
